### ISSUE TITLE
Update dealerdirect/phpcodesniffer-composer-installer dependency to PHP 8 compatible version

### ISSUE THREAD

**Issue by sivaschenko** [magento/magento2#31175]
### Description (*)
Update dealerdirect/phpcodesniffer-composer-installer dependency version from "^0.5.0" to "^0.7.0"
Ensure `composer install` runs on PHP 7.4 and 8.0

**Comment by sergeynezbritskiy** [comment]
@magento I am working on this

**Comment by sergeynezbritskiy** [comment]
Hi @sivaschenko I've created the PR to php8-develop branch, but it is impossible to run `composer install` with php8 as per the description to this issue.
Too many packages still require php7, so I'm getting errors like this:
```
Problem 1
    - Installation request for colinmollenhour/php-redis-session-abstract v1.4.3 -> satisfiable by colinmollenhour/php-redis-session-abstract[v1.4.3].
    - colinmollenhour/php-redis-session-abstract v1.4.3 requires php ^5.5 || ^7.0|| ^7.1 || ^7.2 -> your PHP version (8.0.1) does not satisfy that requirement.
  Problem 2
    - Installation request for composer/composer 1.10.17 -> satisfiable by composer/composer[1.10.17].
    - composer/composer 1.10.17 requires php ^5.3.2 || ^7.0 -> your PHP version (8.0.1) does not satisfy that requirement.
  Problem 3
    - Installation request for composer/semver 1.7.1 -> satisfiable by composer/semver[1.7.1].
    - composer/semver 1.7.1 requires php ^5.3.2 || ^7.0 -> your PHP version (8.0.1) does not satisfy that requirement.
  Problem 4
    - Installation request for elasticsearch/elasticsearch v7.7.0 -> satisfiable by elasticsearch/elasticsearch[v7.7.0].
    - elasticsearch/elasticsearch v7.7.0 requires php ^7.1 -> your PHP version (8.0.1) does not satisfy that requirement.
...
```

**Comment by sivaschenko** [comment]
Hi @sergeynezbritskiy thanks for the PR! We will not be able to test Magento on PHP 8 until all dependencies are updated, at the current stage we need to ensure the new package versions are functional on currently supported by Magento PHP versions `~7.3.0||~7.4.0`

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko. 

Thank you for your report and collaboration!

The related internal Jira ticket `MC-39644` was closed as `Complete`. 

The fix will be available with the upcoming `2.4.3` release.
